components/world_news_api/sources/news-event/news-event.mjs

```diff
@@ -0,0 +1,177 @@
+      propDefinition: [
+        app,
+        "maxSentiment",
+      ],
+    },
+    earliestPublishedDate: {
+      propDefinition: [
+        app,
+        "earliestPublishedDate",
+      ],
+    },
+    latestPublishedDate: {
+      propDefinition: [
+        app,
+        "latestPublishedDate",
+      ],
+    },
+    newsSources: {
+      propDefinition: [
+        app,
+        "newsSources",
+      ],
+    },
+    authors: {
+      propDefinition: [
+        app,
+        "authors",
+      ],
+    },
+    entities: {
+      propDefinition: [
+        app,
+        "entities",
+      ],
+    },
+    locationFilter: {
+      propDefinition: [
+        app,
+        "locationFilter",
+      ],
+    },
+  },
+  methods: {
+    emit(data) {
+      this.$emit(data, {
+        id: data.id,
+        summary: data.title,
+        ts: Date.now(),
+      });
+    },
+    setLastId(id) {
+      this.db.set("lastId", id);
+    },
+    getLastId() {
+      return this.db.get("lastId");
+    },
+    async getCurrentPageRawData(offset, itemsPerPage, $) {
+      const params = {
+        "text": this.text || undefined,
+        "source-countries": getCommaSeparatedListFromArray(this.sourceCountries),
+        "language": this.language || undefined,
+        "min-sentiment": this.minSentiment || undefined,
+        "max-sentiment": this.maxSentiment || undefined,
+        "earliest-published-date": this.earliestPublishedDate || undefined,
+        "latest-published-date": this.latestPublishedDate || undefined,
+        "news-sources": getCommaSeparatedListFromArray(this.newsSources),
+        "authors": getCommaSeparatedListFromArray(this.authors),
+        "entities": getCommaSeparatedListFromArray(this.entities),
+        "location-filter": this.locationFilter || undefined,
+        "offset": offset,
+        "number": itemsPerPage,
+        "sort": "publish-time",
+        "sort-direction": "DESC",
+      };
+      const res = await this.app.searchNews(params, $);
+      return res;
+    },
+    getCurrentPageNewsArray(res, lastEmmitedId) {
+      const news = [];
+      let foundLastEmmitedId = false;
+      for (const news of res.news) {
+        if (news.id === lastEmmitedId) {
+          foundLastEmmitedId = true;
+          break;
+        }
+        news.unshift(news);
+      }
+
+      return {
+        news,
+        foundLastEmmitedId,
+      };
+    },
+  },
+  async run({ $ }) {
+    const ITEMS_PER_PAGE = 100;
+    const MAX_CALLS_PER_EXECUTION = 10;
+    const MAX_OFFSET = MAX_CALLS_PER_EXECUTION * ITEMS_PER_PAGE;
+
+    const lastEmmitedId = this.getLastId();
+    const newsToEmit = [];
+    let offset = 0;
+
+    do {
+      const res = await this.getCurrentPageRawData(offset, ITEMS_PER_PAGE, $);
+      if (res.news.length === 0) {
+        break;
+      }
+
+      const {
+        news,
+        foundLastEmmitedId,
+      } = this.getCurrentPageNewsArray(res, lastEmmitedId);
+      if (foundLastEmmitedId) {
+        break;
+      }
```

Comment:
If `foundLastEmmitedId === true && news.length > 0`, wouldn't this `break` skip the `news` that should be emitted?

Reply:
If I continue after finding the lastEmmitedId, I would emit duplicate emitted events. Even though using dedupe. Dedupe just looks for the last 100 events.